Database_Task.txt

```diff
@@ -1 +1,124 @@
+-- Created DATABASE employee
+CREATE DATABASE employee;
 
+-- Created table hobby
+CREATE TABLE hobby (
+   id INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+   NAME VARCHAR(50) NOT NULL
+);
+
+-- Created table employee
+CREATE TABLE employee (
+   id INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+   first_name   VARCHAR(50)   NOT NULL,
+   last_name VARCHAR(50) NOT NULL,
+   age TINYINT(5) NOT NULL,
+   mobile_number VARCHAR(17) NOT NULL,
+   address VARCHAR(100) NOT NULL
+);
+
+-- Created table employee_salary
+CREATE TABLE employee_salary (
+   fk_employee_id INT NOT NULL,
+   salary DECIMAL(10,2) NOT NULL,
+   DATE DATE NOT NULL,
+   FOREIGN KEY (fk_employee_id) REFERENCES employee(id)
+);
+
+-- Created table employee_hobby
+CREATE TABLE employee_hobby (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    fk_employee_id INT NOT NULL,
+    fk_hobby_id INT NOT NULL,
+    FOREIGN KEY (fk_employee_id) REFERENCES employee(id),
+    FOREIGN KEY (fk_hobby_id) REFERENCES hobby(id)
+);
+
+-- Inserted data in employee table
+INSERT INTO employee (first_name, last_name, age, mobile_number, address) 
+VALUES ("Mashuk", "Patel", 22, "9662321092", "Himatnagar"),
+       ("Rohit", "Vanzara", 22, "8650012345", "Idar"),
+       ("Saurav", "Ninama", 23, "9178647893", "Motipura");
+
+-- Inserted data in employee_salary table
+INSERT INTO employee_salary (id, salary, DATE) 
+VALUES (1, "12345", "2024-05-24"),
+       (2, "54321", "2024-05-01"),
+       (3, "7890", "2023-02-24");
+
+-- Inserted data in hobby table
+INSERT INTO hobby (id,NAME) 
+VALUES (1, "Cricket"),
+       (2, "Football"),
+       (3, "Kabbadi");
+
+-- Inserted data in employee_hobby table
+INSERT INTO employee_hobby (employee_id, hobby_id)
+VALUES (1, 1),
+       (1, 3),
+       (2, 2),
+       (3, 1);
+
+
+-- Updated employee data
+UPDATE employee
+SET first_name = 'Rahul'
+WHERE id = 1;
+
+-- Deleted one employee data
+DELETE FROM employee WHERE id = 1;
+
+-- Truncate table employee
+TRUNCATE TABLE employee_hobby;
+
+-- Create a separate select queries to get a hobby, employee, employee_salary, employee_hobby.
+
+   -- Select query for hobby table
+   SELECT * FROM hobby;
+
+   -- Select query for employee table
+   SELECT * FROM employee;
+
+   -- Select query for employee_salary table
+   SELECT * FROM employee_salary;
+
+   -- Select query for employee_hobby table
+   SELECT * FROM employee_hobby;
+
+
+-- Create a select single query to get all employee name, all hobby_name in single column
+SELECT 
+   CONCAT(e.first_name, ' ', e.last_name, ' - ', GROUP_CONCAT(h.name SEPARATOR ', ')) AS employee_and_hobbies
+FROM 
+   employee e
+LEFT JOIN 
+   employee_hobby eh ON e.id = eh.fk_employee_id
+LEFT JOIN 
+   hobby h ON eh.fk_hobby_id = h.id
+GROUP BY 
+   e.id;
+
+-- Create a select query to get employee name, his/her employee_salary
+SELECT 
```

Comment:
What will happen when there are multiple salary records with the same emp name in this scenario? As per your query, it displays only match records but not display the another emp record